simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -84,8 +78,7 @@ protected ByteArrayOutputStream initialValue()
         public void handle(HttpExchange t) throws IOException {
             String query = t.getRequestURI().getRawQuery();
             String contextPath = t.getHttpContext().getPath();
-            ByteArrayOutputStream response = this.response.get();
-            response.reset();
+            ByteArrayOutputStream response = new ByteArrayOutputStream(1 << 20);
```

Comment:
If I get correctly this comment from @fstab (https://github.com/prometheus/client_java/issues/703#issuecomment-930470387)

>If the initial 1M capacity is not enough, the LocalByteArray will increase in size and copy the data to a larger array. Keeping it will have the effect that this happens only once, because after that it remains in memory with the new capacity.

the previous implementation had the advantage to avoid enlarging the array for each request if the data is larger than 1M. 

If I may suggest a possible approach to keep this feature, we could use an `AtomicInteger` to keep track of the maximum size of the serialized requests so far, and initialize the `ByteArrayOutputStream` immediately to the required size (maybe adding an extra 1K, to avoid expanding the array if the array grows of few bytes). 
This would also save us some byte if the request is smaller than 1M.

Reply:
`LocalByteArray` (i.e. the `ByteArrayOutputStream`) will always grow to the maximum number of bytes that have been written using the object. Being `ThreadLocal` vs method local doesn't change the behavior.

Reply:
I'm sorry @dhoard , my comment was not clear (or I might have misinterpreted @fstab comment). Let me try to explain better my point.

The `ByteArrayOutputStream` is initialized to have, internally, a byte array with initial capacity of 1M. If the response written in the `ByteArrayOutputStream` is larger than the capacity, then the underlying byte array is automatically enlarged to fit the data (typically doubling the size, from 1M to 2M). This enlargement does not come completely for free, since a new array needs to be allocated and the data must be copied from one array to the other.

The previous implementation reduced the number of times that the enlargement was performed, by keeping in memory, and reusing, the instance of `ByteArrayOutputStream` that was already been enlarged.
For example, if you had to write responses of 1.5M, each instance of `ByteArrayOutputStream` was enlarged only once for the entire life of the application.

After the proposed change, instead, every time a request is processed a new `ByteArrayOutputStream` is created with initial capacity equals to 1M. If the serialized response is bigger than the 1M capacity, therefore, the enlargement need to be performed for each request.

Assuming that this additional processing may be a problem - but I would leave to @fstab to comment on whether this is important or not - I was suggesting a way to avoid the array enlargement to happen too frequently, which is to keep track of the required capacity and use this information to initialize the newly created `ByteArrayOutputStream`.